### PR TITLE
Document nft_info command

### PR DESCRIPTION
**Pull request by amarantha-k**

### CODE REVIEW

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (ledhed2222) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: this should always be present, never omitted. it represents the most recent ledger <= the requested ledger that this token was modified. modified here means any of the following:

1) minted (created)
2) ownership changed
3) burned (destroyed)

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (ledhed2222) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: does not return the URI

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (ledhed2222) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: we could also add something like "if you need to retrieve the URI of a burnt token, re-request `nft_info` for this token, specifying the `ledger_index` one previous to the index where this token was burned."

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (ledhed2222) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: looks like there's an errant leading space here prior to `nft_sequence`

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (ledhed2222) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: We should also add some clarity on how transfer fees are represented. See the existing documentation for `NFTokenMint` transaction

**Review by ledhed2222 — CHANGES_REQUESTED** [review]
see comments - thank you!

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (ledhed2222) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: users cannot use `ledger_index: current` for this request!

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (ledhed2222) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: there is no command line option for this

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (ledhed2222) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: we need to call out that users cannot use `ledger_index: current` for this request. doing so will forward the request to `rippled`, but `rippled` does not know this API.

Comment (ledhed2222) [reply content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: also cannot use `closed` as a value here. can only use `validated` or a ledger number

Comment (ledhed2222) [reply content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: we could also say that not specifying `ledger_hash` or `ledger_index` is equivalent to passing `validated`

**Comment by amarantha-k** [comment]
Incorporated review comments. PTAL

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (ledhed2222) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: Specifying `current` _or_ `closed` will result in this. This wording is unclear to me WRT the behavior of `closed`.

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (ledhed2222) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: `warnings` is up one level in the next release. IMO we should just document that since it will be released soon. trying to determine when that will be. here's an example that we can use if so:


```
{
  "result": {
    "nft_id": "00080000B4F4AFC5FBCBD76873F18006173D2193467D3EE70000099B00000000",
    "ledger_index": 269,
    "owner": "rG9gdNygQ6npA9JvDFWBoeXbiUcTYJnEnk",
    "is_burned": false,
    "flags": 8,
    "transfer_fee": 0,
    "issuer": "rHVokeuSnjPjz718qdb47bGXBBHNMP3KDQ",
    "nft_taxon": 0,
    "nft_sequence": 0,
    "uri": "https://xrpl.org",
    "validated": true,
    "status": "success"
  },
  "warnings": [
    {
      "id": 2001,
      "message": "This is a clio server. clio only serves validated data. If you want to talk to rippled, include 'ledger_index':'current' in your request"
    },
    {
      "id": 2002,
      "message": "This server may be out of date"
    }
  ]
}
```

Comment (ledhed2222) [reply content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: that response is JSON RPC - here's websockets:

```
{
  "id": 1,
  "result": {
    "nft_id": "00080000B4F4AFC5FBCBD76873F18006173D2193467D3EE70000099B00000000",
    "ledger_index": 270,
    "owner": "rG9gdNygQ6npA9JvDFWBoeXbiUcTYJnEnk",
    "is_burned": true,
    "flags": 8,
    "transfer_fee": 0,
    "issuer": "rHVokeuSnjPjz718qdb47bGXBBHNMP3KDQ",
    "nft_taxon": 0,
    "nft_sequence": 0,
    "validated": true
  },
  "status": "success",
  "type": "response",
  "warnings": [
    {
      "id": 2001,
      "message": "This is a clio server. clio only serves validated data. If you want to talk to rippled, include ledger_index:current in your request"
    },
    {
      "id": 2002,
      "message": "This server may be out of date"
    }
  ]
}
```

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (ledhed2222) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: `info` object? there's no `info` key. do you just mean that the response is itself an "nft_info object" by virtue of the fact that it is an object and is the response to `nft_info`?

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (ledhed2222) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: missing a space before "(created)"

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (ledhed2222) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: simpler wording suggestion "The account ID of this NFT's owner at the above ledger index."

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (DennisDawson) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: Suggest "the [NFT]" rather than including _the_ in the link.

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (DennisDawson) [reply content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: Caution on predicting the future - we should document how things work when the doc is published. If there's going to be an imminent change, I suggest holding the publication until the feature is live. Otherwise, it's best to document reality today and update it when the change is live.

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (DennisDawson) [reply content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: Perhaps refer to the `info` object as the "`nft_info` response."

**Review by DennisDawson — APPROVED** [review]
Consider the suggestions, but otherwise this looks good to go.

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (amarantha-k) [reply content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: +1 to that. I propose handling this in a separate PR that can be merged once the next release is live.

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (amarantha-k) [reply content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: Yes, will update the "info" object to "nft_info" response.

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (mDuo13) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: This link won't work, currently, and should be removed. (For it to work, we need to add an example command to the `assets/js/apitool-methods-ws.js` file, and also have a publicly available Clio server that the WebSocket Tool can access. When I tested just now, even the NFT-Devnet servers don't currently support `nft_info`.)

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (mDuo13) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: I still don't think "Look up info about a given NFT" belongs in "Server Info" (which is for methods that are mostly about checking the status of / debugging the server itself).

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (mDuo13) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: I'm not sure "human readable" is the right way to describe this data. I mean, sure, it's more readable than a raw NFT ID is but it's still JSON. You could sidestep the question by simplifying this (also, the one in the metadata):

```suggestion
The `nft_info` command asks the Clio server for information about the [NFT](non-fungible-tokens.html) being queried. 
```

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (mDuo13) [reply content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: I agree that we document how it works _now_, which means we should still replace the example response with an actual WebSocket response (the example here is a JSON-RPC response and it's missing the opening/closing braces too)

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (mDuo13) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: "NULL" does not need to be in all caps, but the **Type** column should probably read `String or null` if this can actually return null.

**Review by mDuo13 — CHANGES_REQUESTED** [review]
Don't forget to add the page to the `dactyl-config.yml` file before merging it. Otherwise this file will be ignored and not built as part of the site.

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (amarantha-k) [reply content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: I was looking for a better place for this file and hence haven't added it to the dactyl-config.yml file yet. None of the existing sections seem to be a good fit. Maybe create a new section "Tokenization methods" or "Misc Methods"?

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (cjcobb23) [reply content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: The release is imminent, the code has already been merged to develop. Please update the documentation to what is in develop

Comment (cjcobb23) [reply content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: We can hold this documentation until the release itself is live, which will be very soon

**Review by ledhed2222 — APPROVED** [review]

File: dactyl-config.yml
Comment (mDuo13) [inline dactyl-config.yml]: By convention, untranslated pages should be included in the Japanese target if there's no translated version. This might fix the link checking error you're seeing with the 404 page.

```suggestion
        targets:
            - en
            - ja

    -   md: references/http-websocket-apis/public-api-methods/clio-methods/nft_info.md
        targets:
            - en
            - ja
```

File: content/references/http-websocket-apis/public-api-methods/public-api-methods.md
Comment (mDuo13) [inline content/references/http-websocket-apis/public-api-methods/public-api-methods.md:100]: Suggestion: add the Clio `server_info` here (maybe in addition to the above "Server Info Methods" list, which currently doesn't have it either).

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md
Comment (amarantha-k) [reply content/references/http-websocket-apis/public-api-methods/server-info-methods/nft_info.md]: https://github.com/XRPLF/xrpl-dev-portal/pull/1462 has been created to capture the imminent change.

File: content/references/http-websocket-apis/public-api-methods/clio-methods/nft_info.md
Comment (mDuo13) [inline content/references/http-websocket-apis/public-api-methods/clio-methods/nft_info.md]: You could add a badge to mark when this feature was added and link the Clio release notes like this:

```suggestion
The `nft_info` command asks the Clio server for information about the [NFT](non-fungible-tokens.html) being queried. [New in: Clio v1.1.0](https://github.com/XRPLF/clio/releases/tag/1.1.0 "BADGE_BLUE")
```

(The version number and URL are my speculation. They don't currently exist, but if 1.1.0 is in fact the version the link should start working when the release is tagged on GitHub.)

File: content/references/http-websocket-apis/public-api-methods/clio-methods/nft_info.md
Comment (mDuo13) [inline content/references/http-websocket-apis/public-api-methods/clio-methods/nft_info.md:44]: For what it's worth, I put together a test commit ( https://github.com/mDuo13/xrpl-dev-portal/commit/735656cedb93405770bb4eb52694f8f18174cd9b ) that (a) adds the nft_info command to the WS tool methods, and and (b) adds a public Clio server on NFT-Devnet to the tool. This is **just an example** to show loosely what needs to be done on the tool side to make the "Try it!" work. I don't think we are sure that this particular Clio URL will be long-term supported, though, so **do not merge** my commit.

**Review by mDuo13 — APPROVED** [review]
I think the warning level change should be in this PR (since it's going out in the same Clio release as this API method), and the change to add the new files to the Japanese target should go in before merging. Otherwise, this looks good to go (when the Clio release is live)

File: content/references/http-websocket-apis/public-api-methods/public-api-methods.md
Comment (amarantha-k) [reply content/references/http-websocket-apis/public-api-methods/public-api-methods.md:100]: Yes, will make that change in the other PR where the server_info method is being added.

**Comment by amarantha-k** [comment]
Since the warning level change is also part of the same release, merged the changes from PR #1462 here.

**Comment by amarantha-k** [comment]
Merged as part of #1464

**Pull request closed without merging**
